desktop/src/navigator.rs

```diff
@@ -48,11 +48,16 @@ impl ExternalNavigatorBackend {
 
         let client = builder.build().ok().map(Rc::new);
 
+        // We want just the base URL.
+
+        let mut url = base_url;
+        url.path_segments_mut().unwrap().clear();
```

Comment:
I'm afraid we don't want to use `clear`, since then the whole path is removed. Consider:
```rs
let mut url = Url::parse("https://example.com/path/to/game.swf").unwrap();
url.path_segments_mut().unwrap().clear();
println!("{}", url); // "https://example.com/"
```
Instead `.pop_if_empty().pop().push("");` is appropriate, as it preserves the containing path up to the SWF itself:
```rs
let mut url = Url::parse("https://example.com/path/to/game.swf").unwrap();
url.path_segments_mut().unwrap().pop_if_empty().pop().push("");
println!("{}", url); // "https://example.com/path/to/"
```
I think the `.pop_if_empty()` only matters when the SWF URL it actually a directory (i.e. ends with a slash), but I prefer to keep it to be on the safe side.

Reply:
Yup after trying a few things, I agree. ^_^ Let me know what you think @relrelb !